### PR TITLE
Do not change system php ini with xdebug allow

### PR DESCRIPTION
**Pull request by BackEndTea**
This PR:

- [x] Makes sure we don't change the system php ini.

Fixes: #442

### CODE REVIEW

File: src/Command/InfectionCommand.php [hunk 1]
```diff
@@ -274,8 +274,9 @@ private function includeUserBootstrap(InfectionConfig $config): void
 
     private function applyMemoryLimitFromPhpUnitProcess(Process $process, MemoryUsageAware $adapter): void
     {
-        if (\PHP_SAPI === 'phpdbg') {
+        if (\PHP_SAPI === 'phpdbg' || \extension_loaded('xdebug')) {
```
Comment (sidz) [inline src/Command/InfectionCommand.php]: So with this change we will change memory limit only when we use `--coverage` option. Iis that what @sanmai wanted to achieve when he developed that part?

Comment (sanmai) [reply src/Command/InfectionCommand.php]: Normally you're not allowed to change the system php.ini (there are checks to taht). Normally means not running Infection under a root user.

**Comment by sanmai** [comment]
Normally you're not allowed to change the system php.ini (there are checks to that). Normally means not running Infection under a root user. So this naturally makes me think about under which circumstances is this going on?

File: src/Command/InfectionCommand.php (hunk 1 above)
Comment (BackEndTea) [reply src/Command/InfectionCommand.php]: If xdebug handler is active, then xdebug won't be active at this point of the application.

But as per https://github.com/infection/infection/pull/443#issuecomment-412418591, there may be better ways of doind this check

**Comment by BackEndTea** [comment]
This is now part of #448

**Pull request closed without merging**